lib/tasks/sanitize_data.rake

```diff
@@ -0,0 +1,162 @@
+    end
+
+    def sanitize_user(user, sanitized_user)
+      user.email = sanitized_user[:email]
+      user.unconfirmed_email = sanitized_user[:email]
+      user.save!
+    end
+
+    def sanitize_customer(customer, sanitized_user)
+      customer.email = sanitized_user[:email]
+      customer.name = sanitized_user[:first_name]
+      customer.save!
+    end
+
+    def sanitize_tokenized_permission(tokenized_permission)
+      tokenized_permission.token = nil
+      tokenized_permission.save!
+    end
+
+    def sanitize_address(address, sanitized_user)
+      return if address.nil? || address.enterprise.present?
+
+      address.firstname = sanitized_user[:first_name]
+      address.lastname = sanitized_user[:last_name]
+      address.phone = sanitized_user[:phone]
+      address.alternative_phone = sanitized_user[:phone]
+      address.address1 = address.address2 = "Dummy address"
+      address.city = "Dummy City"
+      address.zipcode = "0000"
+      address.company = address.latitude = address.longitude = nil
+      address.save!
+    end
+
+    def sanitize_payment_method(payment_method)
+      payment_method.name = "Payment Method #{payment_method.id}"
+      payment_method.description = payment_method.name
+      payment_method.save!
+    end
+
+    def sanitize_credit_card(card)
+      card.month = card.year = card.start_month = card.start_year = 1
+      card.cc_type = card.first_name = card.last_name = card.last_digits = "Dummy"
+      card.save!
+    end
+
+    def sanitize_payment(payment)
+      payment.response_code = payment.avs_response = payment.cvv_response_code = nil
+      payment.identifier = payment.cvv_response_message = nil
+      payment.save!
+    end
+
+    def sanitize_paypal_express_checkout(paypal_express_checkout)
+      paypal_express_checkout.token = nil
+      paypal_express_checkout.save!
+    end
+
+    def sanitize_stripe_account(stripe_account)
+      stripe_account.stripe_user_id = stripe_account.stripe_publishable_key = nil
+      stripe_account.save!
+    end
+
+    def sanitize_batch(model_class, sanitize_method_name, canned_users = nil)
+      model_class.constantize.all.each_with_index do |object, index|
+        sanitized_user = sanitized_user(canned_users, index) if canned_users
+
+        if sanitized_user
+          puts "Sanitizing #{model_class} #{object.id} with #{sanitized_user[:first_name]}"
+          __send__ sanitize_method_name, object, sanitized_user
+        else
+          puts "Sanitizing #{model_class} #{object.id}"
+          __send__ sanitize_method_name, object
+        end
+      end
+    end
+
+    def sanitized_user(canned_users, index)
+      sanitized_user = canned_users[index % canned_users.size]
+      sanitized_user[:email] = "#{index}_#{sanitized_user[:email]}"
+      sanitized_user
+    end
+
+    def build_canned_users
+      base_canned_users.each do |user|
+        user[:email] = "ofn@example.com"
+        user[:phone] = "0123456789"
+      end
+    end
```

Comment:
Why do we modify these users at runtime with static data? Wouldn't it be simpler to just append it in `#base_canned_users` below?

Reply:
I ended up ditching this approach all together.